### PR TITLE
Ruff checker for Python

### LINKED ISSUE #1974: Add support for `ruff`
Thank you for taking the time to improve Flycheck.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.

## Feature description
Add support for [ruff](https://github.com/charliermarsh/ruff) (An extremely fast Python linter, written in Rust.).

### PR DESCRIPTION
**Pull request by wyuenho**
fixes #1974

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -10688,6 +10689,34 @@ Requires Flake8 3.0 or newer. See URL
...
+(flycheck-define-checker python-ruff
+  "A Python syntax and style checker using the ruff.
+To override the path to the ruff executable, set
+`flycheck-python-ruff-executable'.
+
+See URL `https://beta.ruff.rs/docs/'."
+  :command ("ruff"
+            "check"
+            (config-file "--config" flycheck-python-ruff-config)
+            "--format=text"
```
[the remark this answers is not in the document]
Comment (wyuenho) [reply flycheck.el]: Thank you!

**Comment by idlip** [comment]
For anyone having issue in org src blocks, `org-edit-special` (`C-c '`).

To make it work replace `source-original` with `source-inplace`

Not sure how it affects for python projects.

**Review by waveFrontSet — APPROVED** [review]

**Comment by waveFrontSet** [comment]
My approval is probably not enough to get this merged (I'm just another random dude who wants this merged) but let's at least bump it a little. 🤞

**Comment by wyuenho** [comment]
@bbatsov can this be merged?

**Review by jcs090218 — APPROVED** [review]

**Review by bbatsov — APPROVED** [review]
The code looks good to me.

**Pull request merged**